src/InventoryManager/Handlers/ActionHandler.cs

```diff
@@ -0,0 +1,259 @@
+        ConsoleUI.PromptLine("Deleted successfully", ConsoleColor.Green);
+        ConsoleUI.WaitAndNavigateToMenu();
+        return;
+    }
+
+    /// <summary>
+    /// Handle search actions on the product list.
+    /// </summary>
+    /// <remarks>
+    /// This method prompt the user to give keyword to search and
+    /// pass the keyword to <see cref="ProductList"/> via search function
+    /// and list the matched results, if there is no matches it will inform
+    /// user that there is no matches so navigate to menu. If no products
+    /// are in the list, then it inform the user that there is no products
+    /// in the list so navigate back to menu.
+    /// </remarks>
+    /// <param name="productList">Give access to user list.</param>
+    public static void HandleSearchProduct(ProductList productList)
+    {
+        ConsoleUI.CreateNewPageFor("Search");
+        if (productList.Count() <= 0)
+        {
+            ConsoleUI.PromptLine(ErrorMessages.NoProducts, ConsoleColor.Yellow);
+            ConsoleUI.WaitAndNavigateToMenu();
+            return;
+        }
+
+        string keyword = ConsoleUI.PromptAndGetInput($"Keyword : ");
+        List<Product> products = productList.Search(keyword);
+        if (products.Count == 0)
+        {
+            ConsoleUI.PromptLine(ErrorMessages.NoMatches, ConsoleColor.Yellow);
+            ConsoleUI.WaitAndNavigateToMenu();
+            return;
+        }
+
+        ConsoleUI.CreateNewPageFor("Search results");
+        ConsoleUI.PrintAsTable(products);
+        ConsoleUI.WaitAndNavigateToMenu();
+        return;
+    }
+
+    /// <summary>
+    /// Ask the user for confirmation to exit with a warning of loss of data.
+    /// </summary>
+    /// <returns><see cref="true"/> /if user chooses to exit; otherwise <see cref="false"/></returns>
+    public static async Task<bool> ConfirmExitAsync()
+    {
+        do
+        {
+            string input = ConsoleUI.PromptAndGetInput(ErrorMessages.AppClosingPrompt, ConsoleColor.Magenta);
+            if (input.ToUpper() == "Y")
+            {
+                ConsoleUI.PromptLine("Closing the app...", ConsoleColor.Magenta);
+                await Task.Delay(1000);
+                return false;
+            }
+            else if (input.ToUpper() == "N")
+            {
+                ConsoleUI.CreateNewPageFor("Menu");
+                break;
+            }
+        }
+        while (true);
+        return true;
+    }
+
+    // Consist of helper methods to get values from the user(used only on action handlers)
+    private static class Helper
+    {
+        // Get the index from the user until user enters a valid input.
+        public static int GetIndex(ProductList productList)
+        {
+            int index;
+            do
+            {
+                string indexString = ConsoleUI.PromptAndGetInput("Enter the index of the product : ");
+                if (!int.TryParse(indexString, out index))
```

Comment:
```suggestion
                if (!int.TryParse(indexString, out int index))
```
line 199 can be removed